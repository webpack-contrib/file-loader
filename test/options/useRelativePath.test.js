/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('useRelativePath', () => {
    test('this.options.context', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            useRelativePath: true,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });

    test('options.context', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            context: '/relative/',
            useRelativePath: true,
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });
  });
});
