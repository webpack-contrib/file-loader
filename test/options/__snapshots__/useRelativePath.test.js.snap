// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options useRelativePath options.context 1`] = `"module.exports = __webpack_public_path__ + \\"./9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";"`;

exports[`Options useRelativePath this.options.context 1`] = `"module.exports = __webpack_public_path__ + \\"./9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";"`;
